agent/status_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	serf "github.com/hashicorp/serf/client"
+	. "gopkg.in/check.v1"
+)
+
+func (*AgentSuite) TestSetsSystemStatusFromMemberStatuses(c *C) {
+	resp := &pb.StatusResponse{Status: &pb.SystemStatus{}}
+	resp.Status.Nodes = []*pb.NodeStatus{
+		{
+			MemberStatus: &pb.MemberStatus{
+				Name:   "foo",
+				Status: pb.MemberStatus_Alive,
+				Tags:   tags{"role": string(RoleNode)},
+			},
+		},
+		{
+			MemberStatus: &pb.MemberStatus{
+				Name:   "bar",
+				Status: pb.MemberStatus_Failed,
+				Tags:   tags{"role": string(RoleMaster)},
+			},
+		},
+	}
+
+	setSystemStatus(resp.Status, []serf.Member{{Name: "foo"}, {Name: "bar"}})
+	c.Assert(resp.Status.Status, Equals, pb.SystemStatus_Degraded)
+}
+
+func (*AgentSuite) TestSetsSystemStatusFromNodeStatuses(c *C) {
+	resp := &pb.StatusResponse{Status: &pb.SystemStatus{}}
+	resp.Status.Nodes = []*pb.NodeStatus{
+		{
+			Name:   "foo",
+			Status: pb.NodeStatus_Running,
+			MemberStatus: &pb.MemberStatus{
+				Name:   "foo",
+				Status: pb.MemberStatus_Alive,
+				Tags:   tags{"role": string(RoleNode)},
+			},
+		},
+		{
+			Name:   "bar",
+			Status: pb.NodeStatus_Degraded,
+			MemberStatus: &pb.MemberStatus{
+				Name:   "bar",
+				Status: pb.MemberStatus_Alive,
+				Tags:   tags{"role": string(RoleMaster)},
+			},
+			Probes: []*pb.Probe{
+				{
+					Checker:  "qux",
+					Status:   pb.Probe_Failed,
+					Severity: pb.Probe_Critical,
+					Error:    "not available",
+				},
+			},
+		},
+	}
+
+	setSystemStatus(resp.Status, []serf.Member{{Name: "foo"}, {Name: "bar"}})
+	c.Assert(resp.Status.Status, Equals, pb.SystemStatus_Degraded)
```

Comment:
This should check the full status and not a part of it - or is supposed to exercise just a subset of `setSystemStatus` functionality? The full test is always preferred over the partial one as it will exclude any unexpected changes.
ditto for tests below